In [5]:

library(tidyverse)


players <- read_csv("players.csv")
sessions <- read_csv("sessions.csv")

glimpse(players)
glimpse(sessions)


Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


Rows: 196
Columns: 7
$ experience   <chr> "Pro", "Veteran", "Veteran", "Amateur", "Regular", "Amate…
$ subscribe    <lgl> TRUE, TRUE, FALSE, TRUE, TRUE, TRUE, TRUE, FALSE, TRUE, T…
$ hashedEmail  <chr> "f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8…
$ played_hours <dbl> 30.3, 3.8, 0.0, 0.7, 0.1, 0.0, 0.0, 0.0, 0.1, 0.0, 1.6, 0…
$ name         <chr> "Morgan", "Christian", "Blake", "Flora", "Kylie", "Adrian…
$ gender       <chr> "Male", "Male", "Male", "Female", "Male", "Female", "Fema…
$ Age          <dbl> 9, 17, 17, 21, 21, 17, 19, 21, 47, 22, 23, 17, 25, 22, 17…
Rows: 1,535
Columns: 5
$ hashedEmail         <chr> "bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8a…
$ start_time          <chr> "30/06/2024 18:12", "17/06/2024 23:33", "25/07/202…
$ end_time            <chr> "30/06/2024 18:24", "17/06/2024 23:46", "25/07/202…
$ original_start_time <dbl> 1.71977e+12, 1.71867e+12, 1.72193e+12, 1.72188e+12…
$ original_end_time   <dbl> 1.71977e+12, 1.71867e+12, 1.72193e+12, 1.72188e+

The two data files given for the Minecraft research include information on the players' demographics and activity. I primarily will be using the players.csv, which has on record of 196 unique players and the self-reported characteristics. I will not be using the sessions.csv file which included 1535 gameplay sessions as my question is focusing on newsletter subscription behavior.

| Variable       | Type        | Description                                      |
| -------------- | ----------- | ------------------------------------------------ |
| `hashedEmail`  | ID          | Unique anonymous player ID                       |
| `subscribe`    | Logical     | Whether the player subscribed to the newsletter  |
| `experience`   | Categorical | Self-reported experience level (Beginner → Pro)  |
| `played_hours` | Numeric     | Self-reported hours played before joining server |
| `gender`       | Categorical | Gender identity                                  |
| `Age`          | Numeric     | Player age in years                              |
| `name`         | String      | Optional name field (identifier only)  

Potential Issues:

- Self-reported data may not be fully accurate.
- played_hours is right-skewed with outliers.
- age has NA values.
- Sessions.csv may be needed later for behavioural features.

In [6]:
players |>
  summarize(across(where(is.numeric), ~round(mean(.x, na.rm = TRUE), 2)))


played_hours Age  
1 5.85         21.14

**Broad Question:**
What characteristics or behaviors indicated by the players are the most predictive in producing a subscription to the game-related newsletter?

**Specific Question:**
Using a player's **experience level, age, gender, and prior playing time** can a prediction on whether they subscribe to the newsletter be made?

**Relevance of Data:**
players.csv file uses all these variables which allows for classification modelling